### PR TITLE
feat(microservices): Introduce producerOnlyMode to ClientKafka

### PR DESCRIPTION
**Pull request by esahin90**
This feature flag enables users to skip the consumer group creation and
rebalancing and allows to only act as a producer on Kafka.

Resolves: #7259

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

PR for nestjs-docs: https://github.com/nestjs/docs.nestjs.com/pull/2280

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
A client will always create a consumer with a default group id on kafka, which leads to rebalancing the consumer group and delaying the application. Even if we don't really want to consume any messages, this requires us to create a new consumer group on Kafka.

Issue Number: #7259

## What is the new behavior?
This change introduces a feature flag to ClientKafka for only using a producer and skipping the consumer group registration

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
kafkajs-docs: https://kafka.js.org/docs/consuming
confluent-docs: https://docs.confluent.io/platform/current/clients/consumer.html#consumer-groups

PR for nestjs-docs: https://github.com/nestjs/docs.nestjs.com/pull/2280

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
Instead of introducing breaking changes, can we just add a new flag that will instruct `ClientKafka` to skip the initialization of the `consumer`? Otherwise, even if we decide to merge this change, it would have to wait till v9

**Comment by esahin90** [comment]
hey @kamilmysliwiec
thx for your feedback. This also sound like a good option, will try to adapt the changes

File: packages/microservices/client/client-kafka.ts [hunk 1]
```diff
@@ -100,36 +103,42 @@ export class ClientKafka extends ClientProxy {
...
+    this.producer = this.client.producer(this.options.producer || {});
     await this.producer.connect();
-    await this.consumer.connect();
-    await this.bindTopics();
+
     return this.producer;
   }
 
   public async bindTopics(): Promise<void> {
+    if (!this.consumer) throw Error('No consumer initialized');
```
Comment (kamilmysliwiec) [inline packages/microservices/client/client-kafka.ts:140]:
```suggestion
    if (!this.consumer) {
       throw Error('No consumer initialized');
    }
```

**Comment by esahin90** [comment]
thx for updating & merging the PR

**Pull request merged**